src/modules/ApprovalVotingModule.sol

```diff
@@ -300,57 +281,38 @@ contract ApprovalVotingModule is VotingModule {
         }
     }
 
-    /*//////////////////////////////////////////////////////////////
-                             VIEW FUNCTIONS
-    //////////////////////////////////////////////////////////////*/
-
     /**
-     * @dev Return true if `account` has cast a vote for `proposalId`.
-     *
-     * @param proposalId The id of the proposal.
-     * @param account The address of the account.
-     */
-    function hasVoted(uint256 proposalId, address account) public view override returns (bool) {
-        return _accountVotes[proposalId][account] != 0;
-    }
-
-    /**
-     * @dev Return for, abstain and option votes for a `proposalId`.
+     * Hook called by a governor after execute, for `proposalId` with `proposalData`.
+     * Revert if the transaction has resulted in more tokens being spent than `budgetAmount`.
      *
      * @param proposalId The id of the proposal.
+     * @param proposalData The proposal data encoded as `(ProposalOption[], ProposalSettings)`.
      */
-    function proposalVotes(uint256 proposalId)
-        public
-        view
-        virtual
-        returns (uint256 forVotes, uint256 abstainVotes, uint128[] memory optionVotes)
-    {
-        ProposalVotes memory votes = _proposals[proposalId].votes;
-        return (votes.forVotes, votes.abstainVotes, _proposals[proposalId].optionVotes);
-    }
+    function _afterExecute(uint256 proposalId, bytes memory proposalData) public view override {
+        (, ProposalSettings memory settings) = abi.decode(proposalData, (ProposalOption[], ProposalSettings));
 
-    /**
-     * @dev Used by governor in `_quorumReached`. See {Governor-_quorumReached}.
-     *
-     * @param proposalId The id of the proposal.
-     * @param quorum The quorum value at the proposal start block.
-     */
-    function _quorumReached(uint256 proposalId, uint256 quorum) external view override returns (bool) {
-        _onlyGovernor(_proposals[proposalId].governor);
-        ProposalVotes memory votes = _proposals[proposalId].votes;
+        if (settings.budgetToken != address(0)) {
+            address governor = proposals[proposalId].governor;
 
-        return quorum <= votes.forVotes + votes.abstainVotes;
+            if (
+                proposals[proposalId].initBalance - IERC20(settings.budgetToken).balanceOf(governor)
+                    > settings.budgetAmount
+            ) revert BudgetExceeded();
```

Comment:
This assumes `balanceOf` goes down during the call, but lots of possible situations where it would go up. In that case it'll underflow and revert.

Should explicitly check that it's `less than` and otherwise just pass the check.

Reply:
nice catch! I can wrap this in un unchecked block in that case

Reply:
Oops, nope, I think it's the opposite. If you wrap in unchecked, then `initBalance - balanceOf` will underflow and be a high value. That will set it above the `budgetAmount` and revert.

Reply:
Sorry, poor wording. I meant I can write it like so

```solidity
// If `finalBalance` is higher than `initBalance`, ignore the budget check
if (finalBalance < initBalance) {
    /// @dev Cannot underflow as `finalBalance` is less than `initBalance`
    unchecked {
        if (initBalance - finalBalance > settings.budgetAmount) revert BudgetExceeded();
    }
}
```

Reply:
Gotcha, perfect.